baybe/constraints/discrete.py

```diff
@@ -42,6 +44,24 @@ def get_invalid(self, data: pd.DataFrame) -> pd.Index:  # noqa: D102
         res = reduce(_valid_logic_combiners[self.combiner], satisfied)
         return data.index[res]
 
+    def to_polars(self) -> pl.Expr:
+        """Translate the constraint to Polars expression for filtering.
+
+        Returns:
+            The Polars Expr object to pass as an argument to filter().
+        """
+        satisfied = []
+        for k, cond in enumerate(self.conditions):
+            satisfied.append(cond.to_polars(pl.col(self.parameters[k])))
```

Comment:
I don't understand the logic here, tbh: It seems that the first condition is applied to the first parameter, the second to the second parameter, etc.... Why?

Reply:
this is exactly how this constraint operates: make x conditions re 1 parameter each, chain them via combiner

Reply:
Feel free to resolve